src/zenml/cli/pipeline.py

```diff
@@ -29,6 +30,31 @@
 logger = get_logger(__name__)
 
 
+def _get_module(module: types.ModuleType, config_item: Union[str, dict]) -> Any:
+    """"""
```

Comment:
Missing a docstring here which would be quite useful, as at least I was a little confused what this method was doing at first.

Overall I think this solution should work, however I'm not sure if it's a little too much freedom/options for the user and will end up being confusing. 
If I understood correctly we now we give them the choice of specifying the step/materializer source as:
1) a single string, in which it is expected to be in the pipeline file
2) a dict, which must contain `file` and `name`

Not sure if for example limiting it to option 2 (with the `file` key optional) would be more streamlined and in the end easier to understand, what do you think?

Reply:
Good Points! I have added a docstring. I have also allowed supplying a name without a file to have the same interface even if the step function is in the same file. 

```ỳaml
steps:
  step_2:
    source: my_second_step
```

==

```ỳaml
steps:
  step_2:
    source: 
      name: my_second_step
```

However I think we could leave the other option in without advertising it in the docs? That way be preserve compatibility. What do you think?

Reply:
Sounds good, as long as the code is still understandable

Reply:
I have now removed the ability to supply: 
```yaml
source: step_name
```

If the user passes this he gets an error message that explains the problem well